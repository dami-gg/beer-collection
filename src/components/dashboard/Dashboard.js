// @flow

import React, {Component} from 'react';
import DashboardOption from './dashboard-option/DashboardOption';

import './dashboard.scss';

import crate from '../../assets/images/crate.png';
import tap from '../../assets/images/tap.png';

class Dashboard extends Component {
  render() {
    return (
        <div className="dashboard">
          <DashboardOption
              title="Add beer to your collection"
              url="/beer/add"
              image={tap}
          />
          <DashboardOption
              title="See your collection"
              url="/collection"
              image={crate}
          />
        </div>
    );
  }
}

export default Dashboard;
