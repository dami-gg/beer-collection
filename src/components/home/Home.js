// @flow
import type { Beer } from "../../types/beer.types";

import React, { Component } from "react";
import { connect } from "react-redux";

import Dashboard from "../dashboard/Dashboard";
import { PAGE_OPTIONS } from "./home.constants";
import { RESULTS_PER_PAGE } from "../collection/collection.constants";
import { preloadBeerImage } from "../collection/collection.helpers";

export class Home extends Component {
  props: {
    collection: Array<Beer>
  };

  componentDidUpdate(): void {
    // Preload images for fast load of the first page of the collection in case the user navigates there
    if (
      this.props.collection.length &&
      this.props.collection.length <= RESULTS_PER_PAGE
    ) {
      preloadBeerImage(this.props.collection[this.props.collection.length - 1]);
    }
  }

  render() {
    return (
      <div className="home">
        <Dashboard items={PAGE_OPTIONS} />
      </div>
    );
  }
}

const mapStateToProps = (state: Object) => ({
  collection: state.collection
});

export default connect(mapStateToProps)(Home);
