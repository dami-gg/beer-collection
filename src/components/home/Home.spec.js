import React from 'react';
import {Home} from './Home';
import Dashboard from '../dashboard/Dashboard';

import {StaticRouter as Router} from 'react-router-dom';

import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

describe('Home', () => {
  let component,
      element,
      mock;

  beforeEach(mocks);

  it('should render without crashing', () => {
    shallow(<Home />);
  });

  it('should match the snapshot', () => {
    component = renderer.create(
        <Router context={mock.context}>
          <Home/>
        </Router>
    );
    let json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  describe('render', () => {
    beforeEach(setup);

    it('should render a home element', () => {
      element = component.find('.home');

      expect(element.length).toEqual(1);
    });

    it('should render a Dashboard component', () => {
      element = component.find(Dashboard);

      expect(element.length).toEqual(1);
    });
  });

  function setup() {
    render();
  }

  function render() {
    component = shallow(<Home />);
  }

  function mocks() {
    mock = {
      context: {}
    };
  }
});

