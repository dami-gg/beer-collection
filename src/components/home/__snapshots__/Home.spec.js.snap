// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot 1`] = `
<div
  className="home"
>
  <div
    className="dashboard"
  >
    <div
      className="dashboard__option "
    >
      <a
        className="dashboard__option__card"
        href="/beer/add"
        onClick={[Function]}
      >
        <div
          className="dashboard__option__image-wrapper"
        >
          <img
            alt="logo"
            className="dashboard__option__image"
            src="tap.png"
          />
        </div>
        <div
          className="dashboard__option__title-wrapper"
        >
          <p
            className="dashboard__option__title"
          >
            Add a new beer
          </p>
        </div>
      </a>
    </div>
    <div
      className="dashboard__option "
    >
      <a
        className="dashboard__option__card"
        href="/collection"
        onClick={[Function]}
      >
        <div
          className="dashboard__option__image-wrapper"
        >
          <img
            alt="logo"
            className="dashboard__option__image"
            src="crate.png"
          />
        </div>
        <div
          className="dashboard__option__title-wrapper"
        >
          <p
            className="dashboard__option__title"
          >
            See your collection
          </p>
        </div>
      </a>
    </div>
    <div
      className="dashboard__option "
    >
      <a
        className="dashboard__option__card"
        href="/collection/manage"
        onClick={[Function]}
      >
        <div
          className="dashboard__option__image-wrapper"
        >
          <img
            alt="logo"
            className="dashboard__option__image"
            src="caps.png"
          />
        </div>
        <div
          className="dashboard__option__title-wrapper"
        >
          <p
            className="dashboard__option__title"
          >
            Manage collection
          </p>
        </div>
      </a>
    </div>
    <div
      className="dashboard__option "
    >
      <a
        className="dashboard__option__card"
        href="/statistics"
        onClick={[Function]}
      >
        <div
          className="dashboard__option__image-wrapper"
        >
          <img
            alt="logo"
            className="dashboard__option__image"
            src="statistics.png"
          />
        </div>
        <div
          className="dashboard__option__title-wrapper"
        >
          <p
            className="dashboard__option__title"
          >
            See statistics
          </p>
        </div>
      </a>
    </div>
    <div
      className="dashboard__option dashboard__option--disabled"
    >
      <a
        className="dashboard__option__card"
        href="/map"
        onClick={[Function]}
      >
        <div
          className="dashboard__option__image-wrapper"
        >
          <img
            alt="logo"
            className="dashboard__option__image"
            src="globe.png"
          />
        </div>
        <div
          className="dashboard__option__title-wrapper"
        >
          <p
            className="dashboard__option__title"
          >
            Beer map
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;
